File: 2018_Graduation_project#convert_to_java/TestApp/app/src/main/java/com/android/test/MenuListFragment.java
package com.android.test;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.app.AlertDialog;
import android.app.AlertDialog.Builder;
import android.app.FragmentTransaction;
import android.content.ContentValues;
import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.Bitmap.CompressFormat;
import android.graphics.BitmapFactory;
import android.graphics.drawable.BitmapDrawable;
import android.graphics.drawable.Drawable;
import android.net.Uri;
import android.os.Bundle;
import android.provider.MediaStore;
import android.provider.MediaStore.Images.Media;
import android.view.LayoutInflater;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.app.Fragment;
import android.view.View.OnClickListener;
import android.widget.Adapter;
import android.widget.AdapterView;
import android.widget.EditText;
import android.widget.HorizontalScrollView;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;
import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.OutputStream;
import java.util.ArrayList;
import java.util.HashMap;

import static com.android.test.user_menu_list.GET_FROM_GALLERY;

@SuppressLint({"ValidFragment"})
public final class MenuListFragment extends Fragment {
    private String passName = null;
    private ImageButton ImageTarget;

    public final String getPassName() {
        return this.passName;
    }

    public View onCreateView(LayoutInflater inflater, final ViewGroup container, Bundle savedInstanceState) {
        final View rootView = inflater.inflate(R.layout.menu_fragment, container, false);
        final ListView listView = (ListView)rootView.findViewById(R.id.MenuListView);
        DatabaseHelper DB = new DatabaseHelper(this.getContext());
        ArrayList MenuList = DB.FetchMenu(this.passName);
        listView.setAdapter(new MenuAdapter(MenuList));

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int position, long l) {
                Context context = MenuListFragment.this.getContext();
                Activity activity = (Activity) context;
                final HorizontalScrollView scrollview = (HorizontalScrollView) activity.findViewById(R.id.server_menu_list_tabs);
                TextView menunumber = (TextView) view.findViewById(R.id.MenuNumber);
                final int rowid = Integer.parseInt(menunumber.getText().toString());
                AlertDialog.Builder alert = new AlertDialog.Builder(context);
                alert.setTitle("메뉴 수정");
                alert.setMessage("수정할 메뉴의 정보를 입력해주세요");
                LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);

                View dialogView = inflater.inflate(R.layout.dialog_insert_menu, (ViewGroup) null);
                final EditText menu_name = (EditText) dialogView.findViewById(R.id.input_menu_name);
                final EditText price = (EditText) dialogView.findViewById(R.id.input_menu_price);
                final EditText tab_name = (EditText) dialogView.findViewById(R.id.input_tab_name);
                final ImageButton image = (ImageButton) dialogView.findViewById(R.id.input_image);
                image.setOnClickListener((OnClickListener) (new OnClickListener() {
                    public final void onClick(View it) {
                        Intent intent = new Intent();
                        intent.setType("image/*");
                        intent.setAction(Intent.ACTION_GET_CONTENT);
                        MenuListFragment.this.startActivityForResult(Intent.createChooser(intent, "Select Picture"), GET_FROM_GALLERY);
                    }
                }));
                MenuListFragment.this.ImageTarget = image;

                tab_name.setText(passName);
                TextView temp = (TextView) view.findViewById(R.id.MenuPrice);
                String tempstr = temp.getText().toString();
                price.setText(tempstr.substring(0,tempstr.length()-2));
                temp = (TextView) view.findViewById(R.id.MenuName);
                menu_name.setText(temp.getText().toString());
                Drawable drawable = ((ImageView) view.findViewById(R.id.MenuImageView)).getDrawable();
                if (drawable != null) {
                    image.setImageBitmap(((BitmapDrawable) drawable).getBitmap());
                }

                alert.setView(dialogView);
                alert.setPositiveButton((CharSequence) "수정", (android.content.DialogInterface.OnClickListener) (new android.content.DialogInterface.OnClickListener() {
                    public final void onClick(DialogInterface dialog, int whichButton) {
                        DatabaseHelper DB = new DatabaseHelper(MenuListFragment.this.getContext());
                        ContentValues values = new ContentValues();
                        BitmapDrawable drawable = (BitmapDrawable) image.getDrawable();
                        if (drawable != null) {
                            Bitmap bitmap = drawable.getBitmap();
                            ByteArrayOutputStream stream = new ByteArrayOutputStream();
                            bitmap.compress(CompressFormat.PNG, 100, (OutputStream) stream);
                            byte[] imagex = stream.toByteArray();
                            values.put("Image", imagex);
                        }

                        values.put("TabName", tab_name.getText().toString());
                        values.put("MenuName", menu_name.getText().toString());
                        String price_test = price.getText().toString();
                        String response;
                        if (price_test.length() > 1) {
                            Integer tempint = Integer.parseInt(price_test);
                            values.put("Price", tempint.toString());
                        }

                        response = DB.UpdateMenu(values, rowid);
                        if (response.equalsIgnoreCase("ok")) {
                            Toast.makeText(MenuListFragment.this.getContext(), (CharSequence) "메뉴수정 성공", Toast.LENGTH_SHORT).show();
                            Fragment itemfragment = new MenuListFragment(passName);
                            FragmentTransaction transaction = getFragmentManager().beginTransaction();
                            transaction.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN).replace(R.id.server_menu_list_container, itemfragment).commitAllowingStateLoss();
                        } else {
                            Toast.makeText(MenuListFragment.this.getContext(), (CharSequence) "메뉴수정 실패", Toast.LENGTH_SHORT).show();
                        }

                    }
                }));
                alert.setNeutralButton((CharSequence) "삭제", (android.content.DialogInterface.OnClickListener) (new android.content.DialogInterface.OnClickListener() {
                    public final void onClick(DialogInterface dialog, int whichButton) {
                        DatabaseHelper DB = new DatabaseHelper(scrollview.getContext());
                        String response = DB.RemoveMenu(rowid);
                        if (response.equalsIgnoreCase("ok")) {
                            Toast.makeText(MenuListFragment.this.getContext(), (CharSequence) "메뉴삭제 성공", Toast.LENGTH_SHORT).show();
                            Fragment itemfragment = new MenuListFragment(passName);
                            FragmentTransaction transaction = getFragmentManager().beginTransaction();
                            transaction.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN).replace(R.id.server_menu_list_container, itemfragment).commitAllowingStateLoss();
                        } else {
                            Toast.makeText(MenuListFragment.this.getContext(), (CharSequence) "메뉴삭제 실패", Toast.LENGTH_SHORT).show();
                        }

                    }
                }));
                alert.setNegativeButton((CharSequence) "취소", (android.content.DialogInterface.OnClickListener) null);
                alert.show();
            }
        });

        return rootView;
    }

    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == GET_FROM_GALLERY && resultCode == Activity.RESULT_OK) {
            Uri selectedImage = data.getData();
            Bitmap bitmap = null;
            try {
                bitmap = Media.getBitmap(this.getContext().getContentResolver(), selectedImage);
            } catch (IOException e) {
                e.printStackTrace();
            }
            this.ImageTarget.setImageBitmap(bitmap_rotate.Rotate(bitmap, 90));
        }

    }
    public MenuListFragment(){}
    public MenuListFragment(String TabName) {
        super();
        this.passName = TabName;
    }
}

File: 2018_Graduation_project#android&kotlin/app/src/main/java/com/example/master/a20180609_android_kotlin/MenuListFragment.kt
package com.example.master.a20180609_android_kotlin

import android.annotation.SuppressLint
import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.app.Activity
import android.app.AlertDialog
import android.content.ContentValues
import android.content.Intent
import android.graphics.Bitmap
import android.graphics.drawable.BitmapDrawable
import android.provider.MediaStore
import android.support.design.widget.TabLayout
import android.support.v4.view.ViewPager
import android.widget.*
import java.io.ByteArrayOutputStream


@SuppressLint("ValidFragment")
class MenuListFragment(passedContext: Context,TabName: String) : Fragment(){

    val passThroughContext : Context = passedContext
    val passName = TabName
    private var ImageTarget: ImageButton? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val rootView = inflater!!.inflate(R.layout.menu_fragment, container, false)
        val recyclerView = rootView.findViewById(R.id.MenuRecyclerView) as RecyclerView

        val DB = DatabaseHelper(passThroughContext)

        val MenuList = DB.FetchMenu(passName)

        recyclerView.adapter = MenuAdapter(MenuList)
        recyclerView.layoutManager = LinearLayoutManager(passThroughContext)
        recyclerView.addOnItemClickListener(object : OnItemClickListener {
            override fun onItemClicked(position: Int, view: View) {
                val tabLayout = (passThroughContext as Activity).findViewById<View>(R.id.server_menu_list_tabs) as TabLayout
                val adapter = (passThroughContext as Activity).findViewById<ViewPager>(R.id.server_menu_list_container).adapter
                val rowid = view.findViewById<TextView>(R.id.MenuNumber).text.toString().toInt()
                val alert = AlertDialog.Builder(passThroughContext)
                alert.setTitle("메뉴 수정")
                alert.setMessage("수정할 메뉴의 정보를 입력해주세요")
                val dialogView = layoutInflater.inflate(R.layout.dialog_insert_menu, null)
                val menu_name = dialogView.findViewById<EditText>(R.id.input_menu_name)
                val price = dialogView.findViewById<EditText>(R.id.input_menu_price)
                val tab_name = dialogView.findViewById<EditText>(R.id.input_tab_name)
                val image = dialogView.findViewById<ImageButton>(R.id.input_image)
                image.setOnClickListener {
                    startActivityForResult(Intent(Intent.ACTION_PICK, MediaStore.Images.Media.INTERNAL_CONTENT_URI),GET_FROM_GALLERY)
                }
                ImageTarget = image

                tab_name.setText(adapter!!.getPageTitle(tabLayout.selectedTabPosition))
                price.setText(view.findViewById<TextView>(R.id.MenuPrice).text.toString())
                menu_name.setText(view.findViewById<TextView>(R.id.MenuName).text.toString())
                val drawable = view.findViewById<ImageView>(R.id.MenuImageView).drawable
                if(drawable != null)
                    image.setImageBitmap((drawable as BitmapDrawable).bitmap)

                alert.setView(dialogView)

                alert.setPositiveButton("수정") { dialog, whichButton ->
                    val DB = DatabaseHelper(passThroughContext)
                    var values = ContentValues()

                    val drawable = image.drawable as BitmapDrawable?
                    if(drawable != null){
                        val bitmap = drawable.bitmap
                        val stream = ByteArrayOutputStream()
                        bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream)
                        val image = stream.toByteArray()
                        values.put("Image",image)
                    }

                    values.put("TabName", tab_name.text.toString())
                    values.put("MenuName", menu_name.text.toString())
                    if(price.text.toString().isNotEmpty())
                        values.put("Price", price.text.toString().toInt())
                    var response = DB.UpdateMenu(values,rowid)
                    if(response=="ok") {
                        Toast.makeText(passThroughContext, "메뉴수정 성공", Toast.LENGTH_SHORT).show()
                        adapter!!.notifyDataSetChanged()
                    }else{
                        Toast.makeText(passThroughContext, "메뉴수정 실패",
                                Toast.LENGTH_SHORT).show()
                    }
                }
                alert.setNeutralButton("삭제"){ dialog, whichButton ->
                    val DB = DatabaseHelper(recyclerView.context)
                    var response = DB.RemoveMenu(rowid)
                    if(response=="ok") {
                        Toast.makeText(passThroughContext, "메뉴삭제 성공", Toast.LENGTH_SHORT).show()
                        adapter!!.notifyDataSetChanged()
                    }else{
                        Toast.makeText(passThroughContext, "메뉴삭제 실패",
                                Toast.LENGTH_SHORT).show()
                    }
                }
                alert.setNegativeButton("취소") { dialog, whichButton ->
                }
                alert.show()


            }
        })
        return rootView
    }
    companion object {
        fun newInstance(context: Context, TabName : String): MenuListFragment {
            return MenuListFragment(context,TabName)
        }
    }
    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        //Detects request codes
        if (requestCode == GET_FROM_GALLERY && resultCode == Activity.RESULT_OK) {
            val selectedImage = data!!.data
            val bitmap = MediaStore.Images.Media.getBitmap(passThroughContext.contentResolver, selectedImage)
            ImageTarget!!.setImageBitmap(bitmap.rotate(90))
        }
    }
}